tests/bsim/bluetooth/mesh/CMakeLists.txt

```diff
@@ -2,13 +2,6 @@
 
 cmake_minimum_required(VERSION 3.20.0)
 
-if (NOT DEFINED ENV{BSIM_COMPONENTS_PATH})
```

Comment:
I don't have bsim installed, I don't know how to use it, but from what I can see someone would need to run the ./compile.sh file in tests/bsim, that file has this:
```
: "${BSIM_OUT_PATH:?BSIM_OUT_PATH must be defined}"
: "${BSIM_COMPONENTS_PATH:?BSIM_COMPONENTS_PATH must be defined}"
: "${ZEPHYR_BASE:?ZEPHYR_BASE must be set to point to the zephyr root\
 directory}"
```
So doesn't that mean that you can't really use this with the added find code here because in order to run the .sh file, you must have bsim's environment variables setup before it will even allow the project build commands to be invoked?

Reply:
Yes and no :).
If you want to run the whole regression as in CI using those scripts (which you probably want if you are doing a lot of bsim development), then you still need to set those variables.
If you just want to build individual apps with cmake/ninja or west, you don't have to anymore after this change.

So yes, this change alone does not free users from setting those variables if they want to do everything.
(The instructions still tell people to set those variables:
https://docs.zephyrproject.org/latest/boards/posix/nrf52_bsim/doc/index.html#building-and-running  )

We were hoping to get rid of the compile.sh and run_parallel.sh scripts with twister v2.
So you can consider this PR as just a partial step to remove that requirement (it only removes it in this build part). Which in itself already helps a bit.

Reply:
Oh so you can build the samples normally using cmake? I read the readme which said no twister support so seemingly missed thinking of cmake

Reply:
Yep you can. (you can also use twister mind you, is just that the twister tests are "1 device" tests, and all the tests in the tests/bsim folder are multi device tests)